Load the dataset

In [4]:
import pandas as pd
df=pd.read_csv("/Advanced_pandas_sales_dataset.csv")
df.head()

,Order_ID,Order_Date,Customer,City,Region,Product,Category,Sales,Quantity,Discount,Profit
0,1001,2024-10-01,Priya,Hyderabad,East,Mouse,Accessories,1735,7,0.00,307.52
1,1002,2024-05-03,Sneha,Kolkata,East,Monitor,Electronics,807,2,0.15,49.89
2,1003,2024-01-26,Anita,Mumbai,West,Phone,Electronics,2328,1,0.20,66.47
3,1004,2024-06-06,Sara,Pune,East,Keyboard,Accessories,3933,3,0.20,-71.66
4,1005,2024-10-15,Rahul,Bangalore,South,Monitor,Electronics,3550,2,0.20,146.30


Q1 Dataset Exploration

Perform the following exploratory steps:

Display the first 10 rows

Display the dataset structure using info()

Find the number of rows and columns

Identify the data types of each column

In [13]:
print(f"First 10 rows:\n{df.head(10)}")
print("*"*100)

#Display the dataset structure using info()
df.info()
print("*"*100)

print("\nNumber of rows and columns:")
print(df.shape)
print("*"*100)
print("\nData types of each column:")
df.dtypes

First 10 rows:
   Order_ID  Order_Date Customer       City Region   Product     Category  \
0      1001  2024-10-01    Priya  Hyderabad   East     Mouse  Accessories   
1      1002  2024-05-03    Sneha    Kolkata   East   Monitor  Electronics   
2      1003  2024-01-26    Anita     Mumbai   West     Phone  Electronics   
3      1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4      1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
5      1006  2024-07-04    Pooja  Bangalore   West   Printer       Office   
6      1007  2024-06-24      Dev    Chennai   East   Monitor  Electronics   
7      1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8      1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
9      1010  2024-12-25    Pooja  Ahmedabad  North     Phone  Electronics   

   Sales  Quantity  Discount  Profit  
0   1735         7      0.00  307.52  
1    807         2      0.15   49.89  
2   2328         1  

,0
Order_ID,int64
Order_Date,object
Customer,object
City,object
Region,object
Product,object
Category,object
Sales,int64
Quantity,int64
Discount,float64


Q2 Sales Filtering Analysis

Filter the dataset to display:

Orders where Sales > 3500

Orders where Region = "East" and Sales > 2500

Explain what insights you observe

In [29]:
print("Orders where Sales > 3500:")
sales_above_3500 = df[df["Sales"] > 3500]
print(sales_above_3500)
print("*"*100)


print("\nOrders where Region = 'East' and Sales > 2500:")
east_sales_above_2500 = df[(df["Region"] == "East") & (df["Sales"] > 2500)]
print(east_sales_above_2500)
print("*"*100)

# observing insights
print(f"Total Orders: {len(sales_above_3500)}")
print("*"*100)
print (f"Top Category: {df['Category'].value_counts()}")
print("*"*100)
print(f"City:{df['City'].value_counts()}")
print("*"*100)
print (f"Total sales: {df['Sales'].sum()}")

Orders where Sales > 3500:
     Order_ID  Order_Date Customer       City Region   Product     Category  \
3        1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4        1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
7        1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8        1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
10       1011  2024-06-30     Neha      Delhi  South   Printer       Office   
..        ...         ...      ...        ...    ...       ...          ...   
219      1220  2024-03-22    Meera  Bangalore   West     Phone  Electronics   
222      1223  2024-03-04    Meera  Ahmedabad  South     Phone  Electronics   
223      1224  2024-01-07      Dev  Bangalore   East    Tablet  Electronics   
225      1226  2024-06-14    Anita  Ahmedabad   East    Laptop  Electronics   
229      1230  2024-04-13    Meera    Kolkata  North   Monitor  Electronics   

     Sales  Quantity  Di

Q3 Customer Purchase Analysis

Identify the top 5 customers based on total sales.

Hint: Use GroupBy with aggregation.

In [30]:
#Identify the top 5 customers based on total sales.
top_customers = df.groupby("Customer")["Sales"].sum().nlargest(5)
print(top_customers)

Customer
Nitin    59820
Raj      58416
Rahul    52291
Dev      48500
Isha     46818
Name: Sales, dtype: int64


Q4 Regional Sales Performance

Calculate the total sales generated by each region.

Then identify:

Which region generated the highest revenue?

In [36]:
#total sales generated by each region
total_sales_by_region = df.groupby("Region")["Sales"].sum()
print(total_sales_by_region)
# highest revenue genrated region
highest_revenue_region = total_sales_by_region.idxmax()
print(f"\nRegion with the highest revenue: {highest_revenue_region}")
print(f"\nRegion with the highest revenue amount: {total_sales_by_region.max()}")


Region
East     232675
North    158210
South    183108
West     170227
Name: Sales, dtype: int64

Region with the highest revenue: East

Region with the highest revenue amount: 232675


Q5 Product Performance Analysis

Calculate the total quantity sold for each product.

Which product appears to be the most popular?

In [38]:
#total quantity sold for each product
total_quantity_by_product = df.groupby("Product")["Quantity"].sum()
print(total_quantity_by_product)

# most popular Product
most_popular_product = total_quantity_by_product.idxmax()
print(f"\nMost popular product: {most_popular_product}")


Product
Headphones    115
Keyboard      110
Laptop        140
Monitor       156
Mouse         122
Phone         151
Printer       108
Tablet        150
Name: Quantity, dtype: int64

Most popular product: Monitor


Q6 Multi-Level Grouping

Group the dataset by: Region and Category

Calculate the total sales for each combination.

What insights can you derive from this analysis?

In [39]:
#Group the dataset by: Region and Category and calculating  total sales for each combination
total_sales_by_region_category = df.groupby(["Region", "Category"])["Sales"].sum()
print(total_sales_by_region_category)


Region  Category   
East    Accessories     57646
        Electronics    160907
        Office          14122
North   Accessories     62052
        Electronics     70023
        Office          26135
South   Accessories     80914
        Electronics     86690
        Office          15504
West    Accessories     64811
        Electronics     84128
        Office          21288
Name: Sales, dtype: int64


Insights :
- Across every single region, Electronics generates the highest total sales
- The East region generated the most revenue overall (232,675)
- Office products are consistently the lowest revenue generator across all four
  regions.

Suggestion :
- high stock and marketing focus on Electronics in the East, as this is the
  primary  revenue generater
- More promotional push required for office Supplies as they are lagging
  significantly everywhere.
- We have to apply East strategies in South and West to boost revenue as East
  strategies worked for selling *Electronics*

Q7 Pivot Table Creation

Create a pivot table showing:

Rows → Region

Columns → Product

Values → Total Sales

Which region sells the most laptops?

In [41]:
# Create pivot table
pivot_table = pd.pivot_table(
    df,
    values='Sales',
    index='Region',
    columns='Product',
    aggfunc='sum'
)
print(pivot_table)

print("*"*100)

#  most laptops selling region
max_laptop_selling_region = pivot_table['Laptop'].idxmax()
max_laptop_selling_value = pivot_table['Laptop'].max()

print(f"Region with highest Laptop sales: {max_laptop_selling_region}")
print(f"Total Laptop Sales: {max_laptop_selling_value}")

Product  Headphones  Keyboard  Laptop  Monitor  Mouse  Phone  Printer  Tablet
Region                                                                       
East          23899     10845   55568    31299  22902  39667    14122   34373
North         15450     20538    4540    14922  26064  30535    26135   20026
South         24111     23480   23029    19167  33323  18218    15504   26276
West          17159     38499   10826    18178   9153  27850    21288   27274
****************************************************************************************************
Region with highest Laptop sales: East
Total Laptop Sales: 55568


Q8 Profitability Analysis

Calculate the average profit for each product category.

Which category is most profitable?

In [45]:
#average profit for each product category
avg_profit_category = df.groupby('Category')['Profit'].mean().round(2)
print(avg_profit_category)

# most profitable category
print(f"\n Most profitable category: {avg_profit_category.idxmax()}")

Category
Accessories    410.60
Electronics    396.46
Office         471.94
Name: Profit, dtype: float64

 Most profitable category: Office


Q9 Discount Impact Study

Analyze the relationship between discount and profit.

Tasks:

Calculate the average profit for each discount level

Identify whether higher discounts lead to lower profit

In [56]:

# average profit for each discount level
avg_profit_discount = df.groupby('Discount')['Profit'].mean().round(2).reset_index()
avg_profit_discount['Discount'] = (avg_profit_discount['Discount'] * 100).astype(int).astype(str) + '%'
print(avg_profit_discount)




  Discount  Profit
0       0%  751.21
1       5%  531.37
2      10%  346.46
3      15%  237.19
4      20%   90.03


*Identifying  whether higher discounts lead to lower profit*  
 - We can clearly analyse that As discount:  high , profit : low
 - High discounts → sometimes negative profit (loss)

Strategic Suggestion :    
 - discounts might be necessary to drive sales volume .
 - The company should restrict 15% and 20% discounts
 - If required, reserving them only for extremely high-volume purchases or less profitable items




Q10 End-to-End Data Wrangling Task

Perform the following workflow:

Filter the dataset where Sales > 3000

Group data by Region and Product

Calculate total sales

Reshape the result into a pivot table

Explain the business insights obtained from this analysis.

In [59]:
filtered_df = df[df["Sales"] > 3000]
grouped_sales = filtered_df.groupby(["Region", "Product"])["Sales"].sum().reset_index()
pivot_result = grouped_sales.pivot(index="Region", columns="Product", values="Sales").fillna(0)
print(pivot_result)


Product  Headphones  Keyboard  Laptop  Monitor  Mouse  Phone  Printer  Tablet
Region                                                                       
East          14469      9642   44096    16420  19000  29500     3696   29229
North          5326     15653    4540     5087  22209  24670    17185    3392
South         10026     19249   17950    14921  26928  18218    14143   19613
West          13898     33380    5922    14354   3511  15653    12648   17185


Analysis :  
 - East region shows strong sales in Laptop, Phone and Tablet.
 - West region is strong in Keyboard.
 - South region is strong in Mouse.
 - High-value sales are spread across different products and regions.